File: ai-robot/src/main/java/za/co/sww/rwars/airobot/model/Battle.java
package za.co.sww.rwars.airobot.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record Battle(
        String id,
        String name,
        @JsonProperty("arenaWidth") int arenaWidth,
        @JsonProperty("arenaHeight") int arenaHeight,
        @JsonProperty("robotMovementTimeSeconds") double robotMovementTimeSeconds,
        String state,
        List<Robot> robots,
        List<Object> walls,
        String winnerId,
        String winnerName
) {}

File: robo-demo/src/main/kotlin/za/co/sww/rwars/robodemo/model/Battle.kt
package za.co.sww.rwars.robodemo.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

/**
 * Represents a battle in the Robot Wars game.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
data class Battle(
    val id: String,
    val name: String,
    val arenaWidth: Int = 50,
    val arenaHeight: Int = 50,
    val state: String = "READY",
    val robots: List<Robot> = emptyList(),
    val robotMovementTimeSeconds: Double = 1.0,
    val winnerId: String? = null,
    val winnerName: String? = null,
)
